### PR TITLE
〽️Made mean calculation more precise

### PR DESCRIPTION
**Pull request by JanosGit**

### CODE REVIEW

File: include/vctr/Expressions/BasicMath/Mean.h [hunk 1]
```diff
@@ -211,11 +211,22 @@ class RootMeanSquare : public ExpressionTemplateBase
...
+
+    #if VCTR_USE_GCEM
+        if constexpr (is::realFloatNumber<value_type>)
+        {
+            if (std::is_constant_evaluated())
+            {
+                return value_type (gcem::sqrt (normalizedSum));
+            }
+        }
+#endif
```
Comment (psciri) [inline include/vctr/Expressions/BasicMath/Mean.h]: indentation is not consistent

File: include/vctr/Expressions/BasicMath/Mean.h [hunk 2]
```diff
@@ -211,11 +211,22 @@ class RootMeanSquare : public ExpressionTemplateBase
 
     //==============================================================================
     template <size_t n>
-    VCTR_FORCEDINLINE value_type finalizeReduction (const std::array<value_type, n>& sums) const
+    VCTR_FORCEDINLINE constexpr value_type finalizeReduction (const std::array<value_type, n>& sums) const
     {
         auto sum = n == 1 ? sums[0] : std::reduce (sums.begin(), sums.end());
-
-        return std::sqrt (sum / value_type (src.size()));
+        auto normalizedSum = sum / RealType<value_type> (src.size());
```
Comment (psciri) [inline include/vctr/Expressions/BasicMath/Mean.h]: this is rather ```mean```

**Review by psciri — APPROVED** [review]
approved besides small style remarks

File: include/vctr/Expressions/BasicMath/Mean.h [hunk 3]
```diff
@@ -34,7 +34,7 @@ class Mean : public ExpressionTemplateBase
 
     static constexpr value_type reductionResultInitValue = 0;
 
-    VCTR_FORCEDINLINE void reduceElementWise (value_type& result, size_t i) const
+    VCTR_FORCEDINLINE constexpr void reduceElementWise (value_type& result, size_t i) const
```
Comment (p4th4x0r) [inline include/vctr/Expressions/BasicMath/Mean.h:37]: muchas gracias

Comment (p4th4x0r) [reply include/vctr/Expressions/BasicMath/Mean.h:37]: by the way: i think also the "sum" function could need that extension :)

Comment (JanosGit) [reply include/vctr/Expressions/BasicMath/Mean.h:37]: Good point, added that ;) 

Actually the entire evaluation of reduction expressions was not constexpr ready until now, so I had to add a few more `constexpr` keywords here and there. But it should work now.

**Pull request merged**